plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -368,12 +370,23 @@ public void checkAndSetLatestUpdateCenter() {
      * @param plugins list of plugins to download
      */
     public void downloadPlugins(List<Plugin> plugins) {
-        plugins.parallelStream().forEach(plugin -> {
-            boolean successfulDownload = downloadPlugin(plugin, null);
-            if (!successfulDownload) {
-                throw new DownloadPluginException("Unable to download " + plugin.getName());
+        ForkJoinPool ioThreadPool = new ForkJoinPool(64);
```

Comment:
Doubt we need to calculate some sensible thread pool. 64 seems reasonable for most CPUs that properly intend to run a Jenkins server / Build server.